### PR TITLE
PIE-11 Fixes datetime error

### PR DESCRIPTION
**Pull request by mrzarquon**
Movse from dateutil method to older, more common, datetime implementation

Adds some docs that are also still WIP

### CODE REVIEW

**Comment by mrzarquon** [comment]
Tested manually. Will need to deploy older splunk install to verify that it works also.

File: bin/ta_puppet_report_viewer/modalert_generate_detailed_report_helper.py [hunk 1]
```diff
@@ -1,42 +1,49 @@
...
-
-
 def get_resource_events(uuid, server, token):
   
   # setup the headers
   headers = {'X-Authentication': token}
   #setup the url
   url = 'https://{}:8081/pdb/query/v4'.format(server)
 
+  report_elements = [
```
Comment (bmjen) [inline bin/ta_puppet_report_viewer/modalert_generate_detailed_report_helper.py]: This array of element keys feels like it should be a class variable and made static.

File: bin/ta_puppet_report_viewer/modalert_generate_detailed_report_helper.py [hunk 2]
```diff
@@ -126,14 +134,14 @@ def process_event(helper, *args, **kwargs):
...
+        event_pe_console = event_content['pe_console']
     else:
         event_uuid = None
-        event_puppetdb_callback_hostname = ""
+        event_pe_console = ""
 
     uuid = helper.get_param("transaction_uuid") or event_uuid
 
-    puppetdb_callback_hostname = helper.get_param("puppetdb_callback_hostname") or event_puppetdb_callback_hostname
+    pe_console = helper.get_param("pe_console") or event_pe_console
```
Comment (bmjen) [inline bin/ta_puppet_report_viewer/modalert_generate_detailed_report_helper.py]: Nit alert: In this section, for style purposes, the use of single vs double quotes should be consistent.

File: bin/ta_puppet_report_viewer/modalert_generate_detailed_report_helper.py [hunk 3]
```diff
@@ -142,22 +150,20 @@ def process_event(helper, *args, **kwargs):
...
+
+    for m in detailed_report['metrics']['data']:
+      mdict[m['category']][m['name']] = m['value']
+    
+    detailed_report['metrics'] = dict(mdict)
 
     detailed_report['url'] = 'https://{}/#/inspect/report/{}/events'.format(pe_console, detailed_report["hash"])
 
-    detailed_report["puppetdb_callback_hostname"] = puppetdb_callback_hostname
+    detailed_report["pe_console"] = pe_console
```
Comment (bmjen) [inline bin/ta_puppet_report_viewer/modalert_generate_detailed_report_helper.py]: Nit alert: In this section, for style purposes, the use of single vs double quotes should be consistent.

**Comment by bmjen** [comment]
Left some nits. Also the changes to the ignore, README, etc should be separated into their own commits, as to not get combined with datetime changes if a revert were needed.

**Comment by mrzarquon** [comment]
This is prepped for your merge @bmjen

**Pull request merged**